lib/ash_json_api/json_schema/open_api.ex

```diff
@@ -734,6 +734,7 @@ if Code.ensure_loaded?(OpenApiSpex) do
         "anyOf" => subtypes
       }
       |> unwrap_any_of()
+      |> with_attribute_description(attr)
```

Comment:
I’m not sure about this part because there doesn’t seem to be a test where `resource_attribute_type/2` is used with `Ash.Type.Union`. Writing a test for that felt a bit too heavy for now, so I didn’t go down that route.

That said, since `resource_attribute_type/2` is used in places other than just `resource_write_attribute_type/2`, I thought it made sense to handle the description here as well to avoid it being dropped.

Reply:
We also need to make sure to handle cases where there is no description key as it's not necessarily always there (for certain composed use cases)

Reply:
The with_attribute_description/2 function already takes care of that by handling cases where the description key is missing.

https://github.com/nallwhy/ash_json_api/blob/4f5f13048ed73221544e9fdde8b0475c6efdeda3/lib/ash_json_api/json_schema/open_api.ex#L938-L952